include/ETWProviders/etwprof.h

```diff
@@ -39,32 +39,32 @@ const int kFlagDoubleClick = 100;
 #include <sal.h> // For _Printf_format_string_
 
 // Insert a single event to mark a point in an ETW trace.
-PLATFORM_INTERFACE void ETWMark( const char *pMessage );
+PLATFORM_INTERFACE void ETWMark( _In_z_ PCSTR pMessage );
 // ETWWorkerMark is identical to ETWMark but goes through a different provider,
 // for different grouping.
-PLATFORM_INTERFACE void ETWWorkerMark(const char *pMessage);
+PLATFORM_INTERFACE void ETWWorkerMark(_In_z_ PCSTR pMessage);
 
 // Insert events with one or more generic int or float data fields
-PLATFORM_INTERFACE void ETWMark1I(const char* pMessage, int data1);
-PLATFORM_INTERFACE void ETWMark2I(const char* pMessage, int data1, int data2);
-PLATFORM_INTERFACE void ETWMark1F(const char* pMessage, float data1);
-PLATFORM_INTERFACE void ETWMark2F(const char* pMessage, float data1, float data2);
+PLATFORM_INTERFACE void ETWMark1I(_In_z_ PCSTR pMessage, int data1);
+PLATFORM_INTERFACE void ETWMark2I(_In_z_ PCSTR pMessage, int data1, int data2);
+PLATFORM_INTERFACE void ETWMark1F(_In_z_ PCSTR pMessage, float data1);
+PLATFORM_INTERFACE void ETWMark2F(_In_z_ PCSTR pMessage, float data1, float data2);
 
 // _Printf_format_string_ is used by /analyze
-PLATFORM_INTERFACE void ETWMarkPrintf( _Printf_format_string_ const char *pMessage, ... );
-PLATFORM_INTERFACE void ETWWorkerMarkPrintf( _Printf_format_string_ const char *pMessage, ... );
+PLATFORM_INTERFACE void ETWMarkPrintf( _Printf_format_string_ _In_z_ PCSTR pMessage, ... );

```

Comment:
> but that just means they designed it poorly, IMHO.

That wouldn't surprise me. SAL has a lot of great applications, and lots of potential, but given it's evolution as an internal Microsoft tool, I doubt they got enough useful feedback.